examples/jsm/loaders/GLTFLoader.js

```diff
@@ -3997,7 +3999,21 @@ class GLTFParser {
 
 			parser.associations.get( node ).nodes = nodeIndex;
 
-			return node;
+            if ( nodeDef.skin === undefined ) return node;
+
+			return parser.getDependency( 'skin', nodeDef.skin ).then( function ( skeleton ) {
+
+				node.traverse( function ( mesh ) {
```

Comment:
Are we sure the node won't have descendants that could include their own SkinnedMesh objects here? Wondering if we need to iterate over 1-2 levels of children, instead of doing a full traverse.

Reply:
Yes, I think you are right, thanks.

If I'm right, `node` for this glTF node can form three patterns.

1. `SkinnedMesh` (single primitive mesh)
2. `Group` - `SkinnedMesh` (multiple primitives mesh or single primitive mesh + other type objects eg. camera)
3. `Group` - `Group` - `SkinnedMesh` (multiple primitives mesh + other type objects eg. camera)

Flatted `Group` - `Group` to single `Group` for case 3 at https://github.com/mrdoob/three.js/blob/42c59b4d516fd204f5f760e1f7daefbbbf18c49c/examples/jsm/loaders/GLTFLoader.js#L3922-L3951 ? If we do, we can 

```
const children = node.isGroup ? node.children : [ node ];
for (let i = 0; i < children.length; i++) {
  ...
}
```

Reply:
If we flatten the hierarchy further it'll break tools like https://github.com/pmndrs/gltfjsx that rely on predictable output structure. We _could_ make changes like that but I think we'd want to plan it well ahead. Lately I've been thinking it'd be better if we never auto-flattened the hierarchy and just output `Group - Group - Mesh` even if there are no multi-primitive meshes or other attached objects. Mostly to keep things simpler and more predictable.

Reply:
OK, let me think how to bind elegantly without changing the hierarchy for now... Revisiting the hierarchy is out of scope from this PR.

Reply:
Ah wait, doesn't here run before adding glTF nodes then isn't full traverse good?

Reply:
Ok – I wasn't sure, but if so let's just add a comment and call it good!

Reply:
Adding a comment is a good idea, done, thanks.